backend-lambda/services/productService.js

```diff
@@ -43,7 +43,7 @@ const updateProduct = async (pool, id, product) => {
   const query = format(
     "UPDATE product SET %s WHERE product_id = %L",
     setStrings,
-    id
+    parseInt(id)
```

Comment:
Do you don't want to wait until this late to do the validation because what if id = 'gooblygoo'? What if we refactor `validateProductId` to return the numeric version of `id`. In other words, let's call parseInt there and return it (assuming the results isn't `NaN`).